undertow-servlet/src/main/java/org/wildfly/elytron/web/undertow/server/servlet/ServletSecurityContextImpl.java

```diff
@@ -314,15 +315,15 @@ static class IdentityContainer implements Serializable {
 
         private static final long serialVersionUID = 812605442632466511L;
 
-        private volatile SecurityIdentity securityIdentity;
+        private volatile CachedIdentity securityIdentity;
         private volatile String authType;
 
-        IdentityContainer(final SecurityIdentity securityIdentity, final String authType) {
+        IdentityContainer(final CachedIdentity securityIdentity, final String authType) {
```

Comment:
Would it be possible to make this class as well as its constructor and accessor methods public?
Otherwise, the ProtoStream marshaller for this class needs to resort to lots of ugly reflection.

Reply:
I think it makes sense to make it public then. I updated the PR accordingly